hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/TimeWait.java

```diff
@@ -54,10 +54,11 @@ public static Builder builder() {
   public void waitFor() {
     try {
       if (waitingTime > timeout) {
-        throw new HoodieException("Timeout(" + waitingTime + "ms) while waiting for " + action);
+        LOG.warn("Timeout(" + waitingTime + "ms) while waiting for " + action);
```

Comment:
I have a question: when `waitingTime > timeout`, function `waitFor()` will return without sleep, and will the AbstractStreamWriteFunction::instantToWrite go busy loop waiting for new commit.requested instant?

Reply:
@hbgstc123 +1. And I think if timeout, throwing exception makes sense to me. Otherwise how the timeout works?

Reply:
> I have a question: when `waitingTime > timeout`, function `waitFor()` will return without sleep, and will the AbstractStreamWriteFunction::instantToWrite go busy loop waiting for new commit.requested instant?

In the case of a lot of parallelism, the cost of job restart is also very high.